packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -200,6 +201,12 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
   ///    match the native behavior on that platform.
   GestureVelocityTrackerBuilder velocityTrackerBuilder;
 
+  /// Whether this gesture detector should reject pointers from a mouse device type.
+  ///
+  /// This is used by scrollables to force mouse interactions to go through the
+  /// scroll wheel or scrollbar, instead of allowing drag gestures to move them.
+  final bool rejectMousePointers;
```

Comment:
This could be inverted to something like `touchOnly`?

Reply:
I wonder should this be platform depend? for example set it to true when platform is desktop or web. Do you know what is the android behavior it user uses mouse?

Reply:
This isn't really a platform adaption, its an adaption based on how the user is interacting with the device. My impression of android/ios is that mouse interaction is uncommon, and if it is apps are likely to do the wrong thing anyway.

Reply:
Is the idea that when it is mouse, prevent drag scroll regardless of the platform?

Reply:
Generally speaking scrolling with a mousewheel or scrollbar is more efficient that scrolling by a simulated touch gesture. By enabling mouse based drag in scrollables, flutter forces any gesture accepting widgets within those lists to compete with the scroll drag events. Disambiguation of these gestures isn't perfect and neither are the users movements - this leads to a harder to use application.
 
 For example, try and drag to scroll to the right/left on the flutter dashboard. Because the left/right drags compete with the up/down drags its easy to get stuck accidentally. Of couse that also implies we need provide a way to keep this enabled, to support the nested scrollable with no scrollbar pattern.

Reply:
I also don't think you should try to generalize this much. Its really unlikely that a gesture detector wants to disambiguate between, say, touch and inverse stylus. And if it does, it might be doing something entirely custom which would make extra configurability on our end a waste.

mouse and touch (stylus/inverse stylus) are really very separate kinds of interaction.